xen/arch/arm/coproc/coproc.c

```diff
@@ -353,24 +350,29 @@ struct coproc_device *coproc_alloc(struct platform_device *pdev,
 
     for ( i = 0; i < num_mmios; ++i )
     {
-        res = platform_get_resource(pdev, IORESOURCE_MEM, i);
-        coproc->mmios[i].base = devm_ioremap_resource(dev, res);
+        ret = dt_device_get_address(pdev, i,
+                                    &coproc->mmios[i].addr,
+                                    &coproc->mmios[i].size);
+        if ( ret )
+        {
+            printk(XENLOG_ERR "Unable to retrieve address %u for %s\n",
+                   i, dev_path(coproc->dev));
+            goto out;
```

Comment:
The comment will help in the same way as  "Failed to get irq index 0 for xxx" would explain that you failed to get some irq index.